### PR TITLE
Cloud functions

### PR DESCRIPTION
**Pull request by MadJacksson**

### CODE REVIEW

File: frontend/webpage/src/api/ibm.js [hunk 1]
```diff
@@ -37,6 +37,33 @@ export const ibmAPI = {
             .catch(() => {
                 return {
                     /* unforeseen issues will be caught here */
+                    error: "Something went wrong."
+                }
+            })
+    },
+    updateRoomCount: (roomName, admin, newCountValue) => {
+        console.log("IBMjs" + newCountValue);
```
Comment (virsp) [inline frontend/webpage/src/api/ibm.js]:
```suggestion
```

File: frontend/webpage/src/api/ibm.js [hunk 2]
```diff
@@ -37,6 +37,33 @@ export const ibmAPI = {
...
+        //newCountValue = 15;
+        const path = "/update_room_count" + "?admin=" + admin + "&room_name=" + roomName + "&new_count_value=" + newCountValue
+        console.log("IBM_path"+ path);
```
Comment (virsp) [inline frontend/webpage/src/api/ibm.js]:
```suggestion
```

File: frontend/webpage/src/api/ibm.js [hunk 3]
```diff
@@ -37,6 +37,33 @@ export const ibmAPI = {
...
+        let options = {
+            method: "GET"
+        }
+        return fetch(baseURL + path, options)
+            .then((response) => {
+                if (response.ok) {
+                    console.log("IBM_if");
```
Comment (virsp) [inline frontend/webpage/src/api/ibm.js]:
```suggestion
```

File: frontend/webpage/src/api/ibm.js [hunk 4]
```diff
@@ -37,6 +37,33 @@ export const ibmAPI = {
...
+                    return response.json()
+                }
+                else {
+                    console.log("IBM_else" + newCountValue);
```
Comment (virsp) [inline frontend/webpage/src/api/ibm.js]:
```suggestion
```

File: frontend/webpage/src/api/ibm.js [hunk 5]
```diff
@@ -37,6 +37,33 @@ export const ibmAPI = {
...
+                    return { error: "Connection error." }
+                }
+            })
+            .catch((e) => {
+                console.log("IBM_catch: " + e);
```
Comment (virsp) [inline frontend/webpage/src/api/ibm.js]:
```suggestion
```

File: frontend/webpage/src/components/ApplyChanges.vue [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            this.$parent.$store.dispatch("UPDATE_ROOM_COUNT", arg);
+        },
+        increment() {
+            this.newCountInput +=1
+        },
+        decrement() {
+            this.newCountInput -= 1 
+        },
+        applyAllChanges() {
+            console.log("applyallchanges_"+this.newCountInput);
```
Comment (virsp) [inline frontend/webpage/src/components/ApplyChanges.vue]:
```suggestion
```

File: frontend/webpage/src/main.js [hunk 7]
```diff
@@ -15,7 +15,7 @@ appID.init({
 // Start mirage mocking server if in development mode. aka 'npm run serve'
 if(process.env.NODE_ENV === 'development'){
     console.log("Starting Mirage API mocking");
-    startMirage();
+    //startMirage();
```
Comment (virsp) [inline frontend/webpage/src/main.js]:
```suggestion
    startMirage();
```

File: frontend/webpage/src/views/SearchRoom.vue [hunk 8]
```diff
@@ -36,26 +57,33 @@
...
 import RoomCount from "@/components/RoomCount.vue";
+import ApplyChanges from "@/components/ApplyChanges.vue";
+
 export default {
     components: {
         RoomCount,
+        ApplyChanges,
     },
     data: () => {
+         console.log("AHHH"+RoomCount);
```
Comment (virsp) [inline frontend/webpage/src/views/SearchRoom.vue]:
```suggestion
```

File: frontend/webpage/src/views/SearchRoom.vue [hunk 9]
```diff
@@ -36,26 +57,33 @@
...
             searchInput: "",
+            newCountInput: 0,
         };
     },
     methods: {
         /**
          * Check if a valid search result is present in the store.
          */
         validLastSearchExists() {
+            console.log("validatesearch");
```
Comment (virsp) [inline frontend/webpage/src/views/SearchRoom.vue]:
```suggestion
```

File: frontend/webpage/src/views/SearchRoom.vue [hunk 10]
```diff
@@ -77,6 +105,7 @@ export default {
          * Search for room count information with the user input in the search field.
          */
         searchRoom() {
+            console.log("searchroom -->" + RoomCount.room_count);
```
Comment (virsp) [inline frontend/webpage/src/views/SearchRoom.vue]:
```suggestion
```

**Review by virsp — CHANGES_REQUESTED** [review]
Remove all ```console.log()``` and were golden! Also se my comment on main.js

**Review by virsp — APPROVED** [review]
Works as intended now!

File: frontend/webpage/src/store/index.js [hunk 11]
```diff
@@ -52,6 +52,11 @@ export const actions = {
                 context.commit("setSearchResult", data)
             }))
     },
+    UPDATE_ROOM_COUNT(context,arg) {
+        console.log("UPD_ROOM_CNT" + arg.count);
```
Comment (MadJacksson) [inline frontend/webpage/src/store/index.js]:
```suggestion
```

**Pull request merged**